web/packages/teleterm/src/services/appUpdater/autoUpdatesStatus.ts

```diff
@@ -127,6 +138,24 @@ function findVersionFromClusters(sources: {
   };
 }
 
+/** When `false` is returned, the user will need to click 'Download' manually. */
+export function shouldAutoDownload(updatesStatus: AutoUpdatesEnabled): boolean {
+  if (updatesStatus.source.kind !== 'most-compatible') {
+    return true;
+  }
+
+  return (
+    // If reading a cluster version fails, another cluster might take over
+    // managing updates.
+    // This can trigger an unintended update if the cluster that previously managed updates
+    // is only temporarily unavailable.
+    updatesStatus.source.unreachableClusters.length === 0 &&
+    // If managing cluster exists but is not able to manage the updates,
+    // do not download most compatible version automatically.
+    !updatesStatus.source.skippedManagingClusterUri
```

Comment:
I think it's not clear what "is not able to manage the updates" means. We're handling a very specific edge case here where a cluster was chosen to manage auto-updates but has since turned them off. It might be better to directly state that here.

Edit: Oh, I now see that it can also mean that we were not able to reach that cluster.

Reply:
I specified these states two states explicitly in the comment.